src/variational/advi.jl

```diff
@@ -0,0 +1,213 @@
+using ForwardDiff
+using Flux.Optimise
+
+import Distributions: _rand!
+
+
+function jac_inv_transform(dist::Distribution, x::T where T<:Real)
+    ForwardDiff.derivative(x -> invlink(dist, x), x)
+end
+
+function jac_inv_transform(dist::Distribution, x::Array{T} where T <: Real)
+    ForwardDiff.jacobian(x -> invlink(dist, x), x)
+end
+
+function center_diag_gaussian(x, μ, σ)
+    # instead of creating a diagonal matrix, we just do elementwise multiplication
+    (σ .^(-1)) .* (x - μ)
+end
+
+function center_diag_gaussian_inv(η, μ, σ)
+    (η .* σ) + μ
+end
+
+# Mean-field approximation used by ADVI
+struct MeanField{T, TDists <: AbstractVector{<: Distribution}} <: VariationalPosterior where T <: Real
```

Comment:
I'd say MeanField is not restricted to ADVI. Maybe move this some place else.

Reply:
Agree. I was thinking that when we get more of these "custom" distributions, we move it into a different file, e.g. `distributions.jl` or something. But really no reason not to just do it right away!